cmd/webhook/main.go

```diff
@@ -37,12 +42,47 @@ func main() {
 	webhook.SetupInClusterClient()
 
 	/* register handlers */
+	var httpServer *http.Server
 	http.HandleFunc("/validate", webhook.ValidateHandler)
 	http.HandleFunc("/isolate", webhook.IsolateHandler)
 
-	/* start serving */
-	err := http.ListenAndServeTLS(fmt.Sprintf("%s:%d", *address, *port), *cert, *key, nil)
-	if err != nil {
-		glog.Fatalf("error starting web server: %s", err.Error())
+	/* start webhook server */
+	go func() {
+		for {
+			httpServer = &http.Server{
+				Addr: fmt.Sprintf("%s:%d", *address, *port),
+			}
+			err := httpServer.ListenAndServeTLS(*cert, *key)
+			if err != nil {
+				if err == http.ErrServerClosed {
+					glog.Info("restarting server")
```

Comment:
It's possible that you'll run in to some problems here, because you're immediately re-opening a socket you've just closed, which sometimes the kernel won't let you do. Is there any way you can do the `bind()` and server creation in separate states, and keep the socket open at all times? That will also give you the added benefit of not rejecting connections during reloads.

Reply:
Right. The most 'smart way' to use bind/send/recv for implement https for hot-cert-reload feature.
But for now, for transitional feature (because to implement above takes a long time), the code is necessary, otherwise the admission controller will not work at the time of cert expiration and user need to restart manually.

So let's add it to TODO list (upstream issue) and keep watch. Is that okey?

Reply:
Right. The most 'smart way' to use bind/send/recv for implement https for hot-cert-reload feature.
But for now, for transitional feature (because to implement above takes a long time), the code is necessary, otherwise the admission controller will not work at the time of cert expiration and user need to restart manually.

So let's add it to TODO list (upstream issue) and keep watch. Is that okey?